libcore/src/IO/GeoFileParser.cpp

```diff
@@ -837,10 +837,11 @@ Goal * GeoFileParser::parseWaitingAreaNode(TiXmlElement * e)
     // Additional checks:
     const bool waitPed =
         (wa->GetMinNumPed() > 0 && wa->GetMaxNumPed() > 0 && wa->GetWaitingTime() >= 0.);
-    const bool waitDoor = (wa->GetTransitionID() > 0);
+    const bool waitDoor   = (wa->GetTransitionID() > 0);
+    const bool waitGlobal = (wa->IsGlobalTimer() && wa->GetWaitingTime() >= 0.);
```

Comment:
You could rename the constants to improve readability quite a bit, e.g.:
```cpp
const bool hasValidConstraints = ...
const bool hasValidTransitionId = ...
const bool usesGlobalTimer = ...
if (!hasValidConstraints && !hasValidTransitionId && !usesGlobalTimer)
{
// handle error
}
```

Reply:
done